harkify/lib/viewnotes.dart

```diff
@@ -46,29 +61,53 @@ class _ViewNotesState extends State<ViewNotes> {
     searchFilter = "";
   }
 
+  void voiceHandler(Map<String, dynamic> inference) {
+    if (inference['isUnderstood']) {
+      if (inference['intent'] == 'searchNotes') {
+        print('Searching for: ' + inference['slots']['date']);
+        onSubmitted(inference['slots']['date'].toString());
+      }
+    } else {
+      // TODO handle not inferring
+      print('did not understand');
+    }
+  }
+
+  Future readFilterResults(String theFoundValue) async {
+    if (readResults) {
+      var result = await flutterTts.speak(
+          "Your reminders for " + searchFilter + " are: " + theFoundValue);
+      readResults = false;
+    }
+  }
+
   AppBar buildAppBar(BuildContext context) {
-    return new AppBar(
-        centerTitle: false,
-        title: new Text('View Notes'),
-        actions: [
-          searchBar.getSearchAction(context),
-          Builder(
-            builder: (context) => IconButton(
-              icon: Icon(Icons.menu),
-              onPressed: () => Scaffold.of(context).openEndDrawer(),
-              tooltip: MaterialLocalizations.of(context).openAppDrawerTooltip,
-            ),
-          )
-        ]);
+    return new AppBar(title: new Text('View Notes'), actions: [
+      searchBar.getSearchAction(context),
+      Builder(
+        builder: (context) => IconButton(
+          icon: Icon(Icons.menu),
+          onPressed: () => Scaffold.of(context).openEndDrawer(),
+          tooltip: MaterialLocalizations.of(context).openAppDrawerTooltip,
+        ),
+      )
+    ]);
   }
 
   @override
   Widget build(BuildContext context) {
+    voiceHelper.startPico(voiceHandler);
+
     return FutureBuilder<List<dynamic>>(
         future: textNoteService.getTextFileList(searchFilter),
         builder: (context, AsyncSnapshot<List<dynamic>> textNotes) {
           if (textNotes.hasData) {
+            for (var note in textNotes.data ?? []) {
+              readFilterResults(note.text);
```

Comment:
This method returns a future but isn't getting assigned. I think I am missing something. Also, the method lives in different classes but is it nearly the same? Seems like this can be moved into a service that handles the lookup and returns the future.